example/isolate-service/bin/server.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2018, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Dart implementation of the gRPC helloworld.Greeter server.
+import 'dart:async';
+import "dart:io";
+import "dart:isolate";
+
+import 'package:grpc/grpc.dart';
+
+import 'package:isolate_service/src/generated/helloworld.pb.dart';
+import 'package:isolate_service/src/generated/helloworld.pbgrpc.dart';
+
+
+import "package:isolate/isolate_runner.dart";
+import "package:isolate/ports.dart";
+import "package:isolate/runner.dart";
+import 'package:logging/logging.dart' show Logger, Level, LogRecord;
+
+class GreeterService extends GreeterServiceBase {
+  static final Logger _log = new Logger('GreeterService');
+  @override
+  Future<HelloReply> sayHello(ServiceCall call, HelloRequest request) async {
+    _log.info("sayHello name='${request.name}'");
+    return HelloReply()..message = 'Hello, ${request.name}!';
+  }
+}
+
+
+void initLog([String isolateMarker]) {
+  if (isolateMarker == null){ isolateMarker = "I${Isolate.current.hashCode}"; }
+  Logger.root.level = Level.ALL;
+  Logger.root.onRecord.listen((LogRecord rec) {
+    if (rec.error != null) {
+      print('[$isolateMarker] ${rec.level.name}:${rec.loggerName} ${rec.time}: ${rec.message} ${rec.error}');
+    } else {
+      print('[$isolateMarker] ${rec.level.name}:${rec.loggerName} ${rec.time}: ${rec.message}');
+    }
+  });
+}
+
+final Logger _log = new Logger('main');
+
+Future<void> main(List<String> args) async {
```

Comment:
Could you turn this example into a test?
Probably you can follow the example of round_trip_test.dart to avoid sending protos.

Reply:
I don't understand how to do the correct share port test. Any ideas?